src/main/kotlin/org/elm/lang/core/types/TypeInference.kt

```diff
@@ -624,47 +629,49 @@ private class InferenceScope(
      * @return a pair of the entire declared type (or [TyUnknown] if no annotation exists), and the
      *   number of parameters in the declaration
      */
-    private fun bindParameters(valueDeclaration: ElmValueDeclaration): Pair<Ty?, Int> {
+    private fun bindParameters(valueDeclaration: ElmValueDeclaration): ParameterBindingResult {
         return when {
             valueDeclaration.functionDeclarationLeft != null -> {
                 bindFunctionDeclarationParameters(valueDeclaration, valueDeclaration.functionDeclarationLeft!!)
             }
             valueDeclaration.pattern != null -> {
                 bindPatternDeclarationParameters(valueDeclaration, valueDeclaration.pattern!!)
-                null to 0
+                ParameterBindingResult.Other(0)
```

Comment:
I like this change a lot (changing from using a pair to using a sealed class). Makes the code easier to read. Plus the improved functionality is nice 😄.